clang/lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -151,6 +152,9 @@ static bool useFramePointerForTargetByDefault(const llvm::opt::ArgList &Args,
     }
   }
 
+  if (toolchains::isARMBareMetal(Triple))
```

Comment:
mind adding `&& Triple.getVendor() != Triple::VendorType::Apple` + some tests, so this doesn't change the default for Apple firmwares?

cc @kubamracek

Reply:
Hi @jroelofs, adding the check for Apple shouldn't be necessary. `isARMBareMetal` is trying to stay in it's lane by only returning true if the vendor is `none` and the environment is `eabi` or `eabihf`.

From grepping the LLVM tests I understand the Apple firmware triple would be something like `arm/thumb-apple-none-macho`, correct? I've added a number of tests to cover this.

Reply:
Perhaps renaming `isARMBareMetal` to `isEABIBareMetal` would make things clearer?

Reply:
`isArmEABIBareMetal`?

Reply:
good idea re: `isArmEABIBareMetal`

Reply:
done!